packages/rich-text-types/src/nodeTypes.ts

```diff
@@ -207,11 +207,20 @@ export interface TableCell extends Block {
   /**
    * @minItems 1
    */
-  content: Paragraph[];
+  content: (Paragraph | OrderedList | UnorderedList)[];
 }
 
+// Notice that this is only exported and not reused anywhere else in this file
+// this is because this we only want this interface to generate the schema
+// if we end up decouping the header cell from the regular cell we
+// end up with unwanted side effects when validating the schema in @contentful/validation
 export interface TableHeaderCell extends TableCell {
```

Comment:
Trying to provide some details here on why the TableHeaderCell needs to be coupled with the TableCell.

When doing the schema validation when they are decoupled like in the [previous PR](https://github.com/contentful/rich-text/pull/619) then the validation error message gets split up like this.

```
[
  {
    "name": "in",
    "details": "Value must be one of expected values",
    "path": [
      "content",
      0,
      "content",
      0,
      "content",
      0,
      "nodeType"
    ],
    "value": "paragraph",
    "expected": [
      "table-cell"
    ]
  },
  {
    "name": "in",
    "details": "Value must be one of expected values",
    "path": [
      "content",
      0,
      "content",
      0,
      "content",
      0,
      "nodeType"
    ],
    "value": "paragraph",
    "expected": [
      "table-header-cell"
    ]
  },
  {
    "name": "in",
    "details": "Value must be one of expected values",
    "path": [
      "content",
      0,
      "content",
      0,
      "content",
      0
    ],
    "value": {
      "nodeType": "paragraph",
      "data": {},
      "content": []
    },
    "expected": [
      {
        "$ref": "#/definitions/TableCell"
      },
      {
        "$ref": "#/definitions/TableHeaderCell"
      }
    ]
  }
]
```

which leads to tests failing... just going to leave the coupling for now.

2 options potentially for the future could be:
1. We add some custom parsing for this error message in contentful validations
2. We change the asserts in the tests in the `@contentful/validation` package

Reply:
Not relevant anymore